### DISCUSSION TITLE
Why is IAnnouncer marked obsolete?

### DISCUSSION THREAD

**Discussion by mitselplik** [fluentmigrator/fluentmigrator#1302] [Q&A] [answered]
Call me crazy but right now I see a great deal of value in keeping this interface and the set of classes that are using it.  I want to be able to pipe migration results asynchrously as log messages occur in an IActionResult method of an API to the browser output using a text stream variation of what Stephen Cleary did here: [Streaming Zip on ASP.NET Core](https://blog.stephencleary.com/2016/11/streaming-zip-on-aspnet-core.html)

Logging to the class that uses this interface seems the ideal candidate for accomplishing that goal?

If there is a better approach to streaming the output of a migration to the browser though, I could find no such examples in the documentation.

**Comment by jzabroski** [comment, marked as answer]
Hi Jeremy,

I was not involved when it was marked obsolete, but my guess is that it was over-abstracted.  I've written up thoughts elsewhere in this projects Issues as it relates to bringing back some of the abstraction logic.  I totally get the idea of piping "log events / activities" as data and creating "log messages" from those events / activities.  That is something Its.Log does.  If you search this repos issues for Its.Log I'm sure I've written up my thoughts on it somewhere.  The major downside to making logs an event pipe to a downstream logging framework is that the calls would be virtual, and therefore cannot acquire a lock/synchronization context.
